### DISCUSSION TITLE
Retrieving file URL when uploading files

### DISCUSSION THREAD

**Discussion by kaseyvee** [rustic-ai/ui-components#131] [Ideas]
Hello!

I'd like to open up a discussion regarding the `MultimodalInput` component in our library, particularly concerning the retrieval of URLs for uploaded files. I suppose more specifically the subcomponent `Uploader`. The current implementation doesn't provide a way to retrieve the URL of the uploaded files. I think it's worth exploring some questions to address this.

Some questions/concerns to consider:

1. Can we assume the upload endpoint will return URLs on successful upload?
2. Do we want to wait for a URL before sending a message or allow users to send a message without the URL?
3. Are there any security considerations to take into account when retrieving and handling URLs for uploaded files?

Any insights and suggestions to approaches are welcome and appreciated.

**Comment by lyjeileen** [comment]
Before message is sent, URLs returned by the upload endpoint could be added to the message so that it's available once it's sent. But I think we don't need to wait for a URL to send a message. 
If message is sent before the files are fully loaded (i.e. no urls are returned), server could go through the database and use `messageId` to look for the urls of attached files. Once the urls are found, they could be send as a thread message to the original message to update the file information. Maybe `multipart` component could have a `disableDownload` field if people don't want to support this feature.
